src/mappings/helpers.ts

```diff
@@ -798,3 +807,48 @@ export function createOrLoadSubgraphCategoryRelation(
   }
   return relation as SubgraphCategoryRelation
 }
+
+export function updateCurrentDeploymentLinks(
+  oldDeployment: SubgraphDeployment | null,
+  newDeployment: SubgraphDeployment,
+  subgraph: Subgraph,
+): void {
+  if (oldDeployment != null) {
+    let oldRelationEntity = CurrentSubgraphDeploymentRelation.load(
+      subgraph.currentVersionRelationEntity,
+    )
+    oldRelationEntity.active = false
+    oldRelationEntity.save()
+
+    oldDeployment.activeSubgraphCount = oldDeployment.activeSubgraphCount - 1
+    oldDeployment.save()
+  }
+
+  let newRelationID = newDeployment.id
+    .concat('-')
+    .concat(BigInt.fromI32(newDeployment.subgraphCount).toString())
+  let newRelationEntity = new CurrentSubgraphDeploymentRelation(newRelationID)
+  newRelationEntity.deployment = newDeployment.id
+  newRelationEntity.subgraph = subgraph.id
+  newRelationEntity.active = true
+  newRelationEntity.save()
+
+  newDeployment.subgraphCount = newDeployment.subgraphCount + 1
+  newDeployment.activeSubgraphCount = newDeployment.activeSubgraphCount + 1
+  newDeployment.save()
+
+  subgraph.currentVersionRelationEntity = newRelationEntity.id
```

Comment:
What is the reason for including this field on the subgraph? Just curious

Reply:
So we can update the old deployment active subgraph count when it changes to a new deployment. (that counter basically adds a little bit of extra information so you can know how many subgraphs are actively using that deployment as their current version)